distributed/client.py

```diff
@@ -353,7 +353,11 @@ def type(self):
     def release(self, _in_destructor=False):
         # NOTE: this method can be called from different threads
         # (see e.g. Client.get() or Future.__del__())
-        if not self._cleared and self.client.generation == self._generation:
+        if (
+            not self._cleared
+            and self.client.generation == self._generation
+            and self.client.scheduler is not None
+        ):
             self._cleared = True
             try:
                 self.client.loop.add_callback(self.client._dec_ref, stringify(self.key))
```

Comment:
The _dec_ref method does other things, like reducing the current refcount on the client.  If we're closing then sure, I'm happy for us not to this, but if we're in a state where we might connect again then this seems not quite right.

I'm curious, how does this situation come about?

If we're woried about the scheduler connection being down, then maybe there is a more comprehensive / central place to handle this, like in `_send_to_scheduler`.

Reply:
I believe the alternative is to catch multiple exceptions in `Future.release()`. 
The ignored exception is seen during interpreter shutdown, when it seems the client has closed its comm first. I don't think its been seen in ordinary cleanup of futures.

Reply:
I originally saw this issue in streamz test suite but only when running on DGXs.  I can reproduce on my laptop.  With @martindurant help, I was able to build a [reproducer](https://github.com/dask/distributed/issues/4305) which only relied on dask/distributed but only occurs when exiting the program or closing all the connections